src/ast/node.cr

```diff
@@ -1,117 +1,42 @@
 module Mint
   class Ast
     class Node
-      # Line and column number pair.
-      alias Position = {Int64, Int64}
-
-      struct Location
-        include JSON::Serializable
-
-        # Relative file path of the `Node`,
-        # this `Location` belongs to.
-        getter filename : String
-
-        # Starting line and column number of the `Node`,
-        # this `Location` belongs to.
-        getter start : Position
-
-        # Ending line and column number of the `Node`,
-        # this `Location` belongs to.
-        getter end : Position
-
-        def initialize(@filename, @start, @end)
-        end
-
-        def contains?(line : Int64)
-          start[0] <= line <= end[0]
-        end
-
-        def contains?(line : Int64, column : Int64)
-          case
-          when line == start[0] == end[0] # If on the only line
-            start[1] <= column < end[1]
-          when line == start[0] # If on the first line
-            column >= start[1]
-          when line == end[0] # If on the last line
-            column < end[1]
-          else
-            contains?(line)
-          end
-        end
-      end
+      property from : Parser::Location
+      property parent : Node?
```

Comment:
Shouldn't they be getters instead?

Reply:
No `parent` is set after initialization [here](https://github.com/mint-lang/mint/blob/master/src/parser.cr#L40), and `from` is set in code actions [here](https://github.com/mint-lang/mint/blob/master/src/ls/code_actions/module_actions.cr#L17)